examples/http/fujitsu_irmc_fans_rest_api.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Domotz Custom Driver 
+ * Name: Fujitsu iRMC Fans
+ * Description: Monitors fan status of a Fujitsu iRMC (Integrated Remote Management Controller) device using the Redfish API
+ * 
+ * Communication protocol is HTTPS
+ * 
+ * Tested with Fujitsu iRMC version: S5
+ *
+ * Creates a Custom Driver table with the following columns:
+ *      - Speed: Fan speed measured in RPM
+ *      - Status: Fan status indicating whether the fan is enabled or disabled
+ * 
+ **/
+
+// Function to make an HTTP GET request to retrieve fan status from the Fujitsu iRMC device
+function getFanStatus() {
+    var d = D.q.defer();
+    D.device.http.get({
+        url: "/redfish/v1/Chassis/0/Thermal",
+        username: D.device.username(), 
+        password: D.device.password(), 
+        protocol: "https",
+        auth: "basic",
+        headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        },
+        jar: true,
+        rejectUnauthorized: false
+
+    }, function (error, response, body) {
+        if (error) {
+            console.error(error);
+            D.failure(D.errorType.GENERIC_ERROR);
+        }
+        if (response.statusCode == 401) {
+            D.failure(D.errorType.AUTHENTICATION_ERROR);
+        } else if (response.statusCode != 200) {
+            D.failure(D.errorType.GENERIC_ERROR);
+        }
+        d.resolve(JSON.parse(body));
+    });
+    return d.promise;
+}
+
+// Function to sanitize the record ID 
+function sanitize(output){
+    var recordIdReservedWords = ['\\?', '\\*', '\\%', 'table', 'column', 'history'];
+    var recordIdSanitisationRegex = new RegExp(recordIdReservedWords.join('|'), 'g');
+    return output.replace(recordIdSanitisationRegex, '').slice(0, 50).replace(/\s+/g, '-').toLowerCase();
+}
+
+// Extracts relevant data from the API response and populates the Custom Driver Table
+function extractData(data) {
+    if (!data.Fans || data.Fans.length === 0) {
```

Comment:
if there is no data the process will not exit and it will continue and this will cause an error in line 60, you should fail if there is no data